addons/addon-base/packages/services/lib/user/user-authz-service.js

```diff
@@ -49,6 +51,13 @@ class UserAuthzService extends Service {
   }
 
   // Protected methods
+  async authorizeList(requestContext) {
+    if (requestContext.principal.userRole === 'guest') {
```

Comment:
Is there a differentiation between internal v.s. external guest here?

Reply:
Yes, both are different. @nguyen102 should we add internal guest here as well?

Reply:
Good catch. I updated the code to deny access for internal-guest as well